test/functional/blockprod_generate_pos_blocks.py

```diff
@@ -44,7 +44,10 @@ class GeneratePoSBlocksTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["--chain-pos-netupgrades=true"]]
+        self.extra_args = [[
+            "--chain-pos-netupgrades=true",
+            "--blockprod-min-peers-to-produce-blocks=0",
+        ]]
 
```

Comment:
Btw, the function `assert_pos_consensus()` is giving a warning that `block_hex` is undefined. A language model/server for Python can help to prevent these issues, because it's all at runtime, unlike rust.